compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1070,13 +1070,46 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                 )?
             }
             (
-                &ty::Param(_),
-                Res::SelfTyParam { trait_: param_did } | Res::Def(DefKind::TyParam, param_did),
+                ty::Param(_),
+                Res::SelfTyParam { trait_: param_def_id }
+                | Res::Def(DefKind::TyParam, param_def_id),
```

Comment:
unrelated: are there ever cases where we have a `ty::Param` with a different res? and if so, do we want to support them? we can go from the index of the param to its def_id after all

Reply:
Self params dont resolve to DefKind::TyParam, they resolve to the trait defid and DefKind::Trait iirc